refactor: extract models into separate module to allow for future docify templates to become widgets

LeighFinegold opened the pull request:
## Description

This PR refactors CALM model types into a separate module and plants the seed to migrate advanced widgets from Docify website mode to fully-fledged widgets.

**Problem Being Prevented:**
- Prevents circular dependency that would occur when migrating widgets from `shared` to `calm-widgets` module
- Currently `shared` module contains both CALM model types and widget implementations 
- Moving widgets to `calm-widgets` would create a circular dependency since widgets need access to CALM model types

**Key Changes:**
- Extracted CALM model types from `shared` module into separate `calm-models` module
- Updated dependencies in both `shared` and `calm-widgets` to reference the new `calm-models` module
- Maintained all existing functionality while preparing for future widget migration

**Impact:**
This change unblocks the future migration of advanced widgets (C4 diagrams, flow sequences, related nodes) from Docify website-specific implementations to fully-fledged widgets in the `calm-widgets` module, enabling sophisticated visualization capabilities in all calm-docify output modes.

## Type of Change
<!-- Check the type that applies to your PR -->
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🎨 Code style/formatting changes
- [x] ♻️ Refactoring (no functional changes)
- [ ] ⚡ Performance improvements
- [ ] ✅ Test additions or updates
- [ ] 🔧 Chore (maintenance, dependencies, CI, etc.)

## Affected Components
<!-- Check all that apply -->
- [ ] CLI (`cli/`)
- [x] Shared (`shared/`)
- [x] CALM Widgets (`calm-widgets/`)
- [ ] CALM Hub (`calm-hub/`)
- [x] CALM Hub UI (`calm-hub-ui/`)
- [ ] Documentation (`docs/`)
- [ ] VS Code Extension (`calm-plugins/vscode/`)
- [x] Dependencies
- [ ] CI/CD

## Commit Message Format ✅
<!-- 
Make sure your commit messages follow the conventional commit format:
type(scope): description

Scope is optional but recommended - you can use ./commit-helper.sh to get suggestions!

Examples:
- feat(cli): add new validation command
- fix(shared): resolve schema parsing issue
- docs: update installation guide
- chore(deps): bump typescript to 5.8.3

This helps with our automated versioning and changelog generation!
Note: Only commits with (cli) scope will trigger CLI releases.
-->

## Testing
<!-- Describe how you tested your changes -->
- [x] I have tested my changes locally
- [x] I have added/updated unit tests
- [x] All existing tests pass
- [x] Verified no circular dependency between modules
- [x] Confirmed all existing functionality continues to work
- [x] Validated that both `shared` and `calm-widgets` can now reference `calm-models` independently

## Checklist
- [x] My commits follow the [conventional commit format](https://www.conventionalcommits.org/)
- [x] I have updated documentation if necessary
- [x] I have added tests for my changes (if applicable)
- [x] My changes follow the project's coding standards

LeighFinegold commented on calm-models/src/model/control.ts:
```diff
@@ -5,7 +5,7 @@ import {
     CalmControlDetailCanonicalModel,
     CalmControlsCanonicalModel
-} from '../template/template-models';
+} from '../canonical/template-models.js';
```
Minor: Perhaps should rename this file now that the models is not specifically part of the template engine

LeighFinegold replied:
Will handle in subsequent commit so git diff is easier for dev to see there was no code change but a movement

rocketstack-matt reviewed (dismissed):
The changes make sense, can we change the following before merging:
- [ ] Add the new module to the top level README
- [x] Add a module specific README outlining the purpose and what should(n't) be kept in the models module.

markscott-ms commented on calm-models/package.json:
```diff
@@ -0,0 +1,51 @@
+{
+  "name": "@finos/calm-models",
```
So, this package contains the/a Common Architecture Language **Model model**? Is the package name going to be confusing?

LeighFinegold commented on calm-hub-ui/src/visualizer/components/drawer/Drawer.tsx:
```diff
@@ -3,7 +3,7 @@ import { useCallback, useEffect, useState } from 'react';
     CalmArchitectureSchema,
     CalmNodeSchema,
-} from '../../../../../shared/src/types/core-types.js';
+} from '../../../../../calm-models/src/types/core-types.js';
```
See my comment here - https://github.com/finos/architecture-as-code/pull/1544#discussion_r2296751822. 

I'm not scoping a rewrite of calm-hub-ui dependencies into this change. I maintained the exact reference as of now just with the model files moved from shared to calm-models.

Raised https://github.com/finos/architecture-as-code/issues/1551 for someone else to tackle

LeighFinegold commented on calm-hub-ui/src/visualizer/contracts/contracts.ts:
```diff
@@ -1,6 +1,5 @@
 import cytoscape from 'cytoscape';
-import { CalmInterfaceSchema } from '@finos/calm-shared/src/types/core-types.js';
-import { CalmControlsSchema } from '@finos/calm-shared/src/types/control-types.js';
+import { CalmInterfaceSchema, CalmControlsSchema } from '@finos/calm-models/src/types/control-types.js';
```
Raised https://github.com/finos/architecture-as-code/issues/1551

LeighFinegold commented:
> The changes make sense, can we change the following before merging:
> 
> * [ ]  Add the new module to the top level README
> * [ ]  Add a module specific README outlining the purpose and what should(n't) be kept in the models module.

Added README.md and corrected the root README.md. For now added management under the same maintainers as the cli and shared.
